src/reporter.ts

```diff
@@ -108,6 +123,79 @@ export default class Reporter {
     return job;
   }
 
+  // Reports a spec as a TestRun to Sauce.
+  async reportTestRun(result: RunResult, meta: {jobId: string | undefined}) {
+    const stats = result.stats as RunResultStats;
+    const status = 
+      stats.failures > 0 ? Status.Failed
+        : stats.skipped === stats.tests ? Status.Skipped
+        : Status.Passed;
+
+    const req : TestRunRequestBody = {
+      id: crypto.randomUUID(),
+      name: result.spec.name,
+      start_time: stats.wallClockStartedAt || result.stats.startedAt || '',
+      end_time: stats.wallClockEndedAt || stats.endedAt || '',
+      duration: stats.wallClockDuration ?? result.stats.duration ?? '',
+      platform: 'other',
+      type: 'web',
+      framework: 'cypress',
+      status,
+      errors: this.findErrors(result),
+    };
+
+    if (this.cypressDetails?.browser) {
+      req.browser = `${this.cypressDetails?.browser?.name} ${this.cypressDetails?.browser?.version}`;
+    }
+    if (this.cypressDetails?.system) {
+      req.os = this.getOsName(this.cypressDetails?.system?.osName);
+    }
+    if (this.opts.tags) {
+      req.tags = this.opts.tags?.map((tag) => ({ title: tag }));
+    }
+    // NOTE: If we didn't run in sauce, do we need to define sauce_job for the test run?
```

Comment:
We always ought to have a job ID, apparently. Because we'll always at the very least report a job or actually run on a Sauce VM.